cns/cnsclient/httpapi/client.go

```diff
@@ -11,7 +11,45 @@ type Client struct {
 }
 
 // UpdateCNSState updates cns state
-func (client *Client) UpdateCNSState(createNetworkContainerRequest *cns.CreateNetworkContainerRequest) error {
-	//Mat will pick up from here
-	return nil
+func (client *Client) UpdateCNSState(ipConfigs []*cns.ContainerIPConfigState) error {
+	var (
+		ipConfigsToAdd []*cns.ContainerIPConfigState
+	)
+
+	//Lock to read ipconfigs
+	client.RestService.Lock()
+
+	//Only add ipconfigs that don't exist in cns state already
+	for _, ipConfig := range ipConfigs {
+		if _, ok := client.RestService.PodIPConfigState[ipConfig.ID]; !ok {
+			ipConfig.State = cns.Available
+			ipConfigsToAdd = append(ipConfigsToAdd, ipConfig)
+		}
+	}
+
+	client.RestService.Unlock()
+
```

Comment:
How are you initializing/reconciling the NC state? There could be some stale NC/POD configurations as well, no?